### PR TITLE
Fix short circuit logic for alpha=0 with stream-k

### PR DESCRIPTION
**Pull request by AlexBrownAMD**
This update fixes the case when alpha=0 by ensuring that A/B matrices are not read and main loop does not run. Also added a new small test case with stream-k and alpha=0 to validate output.

### CODE REVIEW

File: tensilelite/Tensile/Components/StreamK.py [hunk 1]
```diff
@@ -1910,8 +1916,22 @@ def graWorkGroup(self, writer, kernel, tPA, tPB):
...
+        # Map SK index to WG
         module.add(self.skIndexToWG(writer, kernel, sTmp))
 
+        # Short circuit if alpha==0 (skip main loop and reading A/B, only do beta * C)
+        # To skip main loop in stream-k, we check if this WG is responsible for writing results (ie: WG starts tile)
+        # If WG starts tile then set LocalEnd=ItersPerTile to skip fixup step, and set loopCounter to 0 to skip main loop
+        # If WG does not start tile, skip to end of persistent loop to check for other SK tile
+        alphaLabel = Label("SKAlphaCheck", "")
+        module.add(SBranchIfNotZero("Alpha", kernel["ProblemType"]["ComputeDataType"], alphaLabel))
+        # Skip to end if not doing the global write
```
Comment (nakajee) [inline tensilelite/Tensile/Components/StreamK.py:1928]: What does this mean?
We need global write even if alpha==0, right?

Comment (AlexBrownAMD) [reply tensilelite/Tensile/Components/StreamK.py:1928]: We need global write of beta * C (unless beta is 0, then it skips that part too)

File: tensilelite/Tensile/Tests/common/streamk/sk_sgemm_alpha0.yaml [hunk 2]
```diff
@@ -0,0 +1,119 @@
+TestParameters:
+  marks: [skip-gfx900, skip-gfx906, skip-gfx1010, skip-gfx1011, skip-gfx1012, skip-gfx1030, skip-gfx1100, skip-gfx1101, skip-gfx1102, skip-gfx1200, skip-gfx1201] # not supported by arch
+
+GlobalParameters:
+  NumElementsToValidate: -1
+  BoundsCheck: False
+  KernelTime: False
+  DataInitTypeAlpha: 0
+  DataInitTypeBeta: 1
```
Comment (nakajee) [inline tensilelite/Tensile/Tests/common/streamk/sk_sgemm_alpha0.yaml]: Maybe better to use Beta !=1 to make sure beta * C is properly stored.

File: tensilelite/Tensile/Components/StreamK.py (hunk 1 above)
Comment (nakajee) [reply tensilelite/Tensile/Components/StreamK.py:1928]: OK. I understand the logic.
StreamKLocalStart=0 WG does global write.
All other WGs with StreamKLocalStart>1, no global write, right?

Comment (AlexBrownAMD) [reply tensilelite/Tensile/Components/StreamK.py:1928]: Yep that's right

**Comment by KKyang** [comment]
https://github.com/ROCm/hipBLASLt/blob/c961e7dbd621ddddd95eb6853e4a992d13dbeedd/tensilelite/Tensile/KernelWriterAssembly.py#L1882

Suggest adding a comment or implement this path to ensure stream-K works with backward pass in the future.

File: tensilelite/Tensile/Tests/common/streamk/sk_sgemm_alpha0.yaml (hunk 2 above)
Comment (AlexBrownAMD) [reply tensilelite/Tensile/Tests/common/streamk/sk_sgemm_alpha0.yaml]: Updated

**Comment by AlexBrownAMD** [comment]
> Suggest adding a comment or implement this path to ensure stream-K works with backward pass in the future.

I don't see any code that needs to be changed in this part right now for stream-k so I added a comment to this section to say how the short-circuit works for the stream-k case

**Review by nakajee — APPROVED** [review]
Looks good

**Review by KKyang — APPROVED** [review]

**Review by neoblizz — APPROVED** [review]
Looks good to me

**Pull request merged**